### ISSUE TITLE
Install Plugin: Improve UI on free plan sites

### ISSUE THREAD

**Issue by taipeicoder** [Automattic/wp-calypso#93606]
## Description

![Image](https://github.com/user-attachments/assets/fa14db45-74bb-4e43-a961-a3fcf5adefcf)

* The layout is confusing with too many boxes, and boxes within boxes.
* The primary action is easy to confuse with "new" label.

**Comment by davemart-in** [comment]
Seems related to https://github.com/Automattic/dotcom-forge/issues/7727

**Comment by davemart-in** [comment]
Also feels related to: https://github.com/Automattic/wp-calypso/issues/92252

**Comment by javierarce** [comment]
> @javierarce just checking in if you think this is an issue you'd like to take a pass at designing a less boxy design for, or if we should just assign it to an engineer and use our best judgement?

I think it would be nice to simplify this. I can work on a proposal for this modal and the one in #7727

**Comment by javierarce** [comment]
I've been reviewing the entire process of uploading a plugin and I was wondering if we could display a modal when the user clicks the Upload button instead of redirecting them to a different screen:

![image](https://github.com/user-attachments/assets/626c6143-91b4-444b-81bb-330e11b7fab5)

This modal would be similar to the one that appears when a user clicks the "Purchase and activate" CTA on a plugin page.

If the user decides to upgrade their plan, we could redirect them to `https://wordpress.com/plugins/upload/{site}` to continue the upload process.

Here are the tweaks I made to the modal. I think we could apply these changes to the other modal as well.

![image](https://github.com/user-attachments/assets/099f2325-ebfa-49be-befc-b944bc63114d)

1. The introductory text informs the user about additional benefits they’ll receive if they upgrade.
2. There’s a more explicit note warning about changes to their domain.
3. The effect of the upgrade to the domain is highlighted in the text.
4. The font size of the "contact support" text is reduced for better visual balance.
5. The badges use the ligther style we developed for the Site Logs, and they should look less like buttons

### Regular flow

For reference, this is a regular flow. We display the same information twice, and I think we could simplify the process and avoid showing the modal again.

https://github.com/user-attachments/assets/c52ba749-2265-40f2-985a-00750a683b67

### Bugs

While testing the upload plugin flow, I detected a couple of problems. The first one happens if there's a third party plugin in the shopping cart.

https://github.com/user-attachments/assets/6a182571-0f9e-4f8f-9707-3be1a90850c6

I've also found a different bug in the flow, but I haven't been able to reproduce it again

https://github.com/user-attachments/assets/92202c00-7b23-4a8b-8bc6-8dd760517587

[Figma file](https://www.figma.com/design/wCm5Tzb7JBaG4WNowWbjES/Logical-Flows?node-id=5085-18)

**Comment by javierarce** [comment]
Here's the mobile version. I did some minor tweaking to fix some problems in the current modal (essentially, reducing some paddings & margins, and making sure the badges don't overlap the domains)

![image](https://github.com/user-attachments/assets/fc2eba80-31cf-4636-a01b-01966b4ed8f4)
![image](https://github.com/user-attachments/assets/345691d7-bf37-4caf-ac51-404a699bd08f)

**Comment by rcrdortiz** [comment]
> @javierarce I have some mixed feelings about using a modal instead of an on-page nudge:
> 
> * Codewise we'll need a conditional to either display the upgrade nudge vs redirecting to the upload screen.
> * It's harder to re-use it in other screens where we have something similar (e.g. Theme upload)
> * It contains a lot of information and warnings upfront that are not needed for the purchase. This might overwhelm the user and decide to not upgrade (e.g. the site address is scary. Although, if they upgrade for a yearly plan they'll get a free domain, so that notice won't actually show again).
> 
> Personally, I prefer the upsell nudge in the Theme upload because it's simple (from Ui and code PoV): click on this and upgrade to be able to upload the theme.
> 
> From stats PoV, a simple upgrade nudge like the one from Theme Upload is more performant than the current one we have in Plugins. Private data shared on slack: p1726148035172049-slack-C04DZ8M0GHW
> 
> ![Screenshot 2024-09-12 at 16 41 28](https://private-user-images.githubusercontent.com/995955/366900982-8a8aabb9-6c73-421c-85ef-f440b598d046.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjYxNDg4ODAsIm5iZiI6MTcyNjE0ODU4MCwicGF0aCI6Ii85OTU5NTUvMzY2OTAwOTgyLThhOGFhYmI5LTZjNzMtNDIxYy04NWVmLWY0NDBiNTk4ZDA0Ni5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwOTEyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDkxMlQxMzQzMDBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iYzVmZDEyNzkyNmU2Nzk5MWY4ZWM5OWU0MmU5Y2E2ODQ5MDQ3NGRmNTE5MWVkNGI0MjJiYzQ4MGI2ZTgwYWI0JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.LPDbUmJKYyzdtO8YIbpcDRYCdQbZr1Suow5x9H5Jaig)

I think we can remove all those messages, especially since we don't display them when entering the theme installation flow. We should make both the "Install Plugin" and "Install Theme" buttons use the same modal. @javierarce, can we come up with a single modal that works for both cases?

**Comment by mmtr** [comment]
I agree with the messages above. I don't think we should show the *.wordpress.com > *.wpcomstaging.com domain change warning before purchase, because that message is scary and users might be less likely to upgrade (and, as pointed out by @BogdanUngureanu we already have evidence that this is case if we compare the theme upload flow with the plugin install flow).

@javierarce can you come up with an alternative design so the upgrade nudge is displayed first, and the domain change warning is displayed after purchase? Ideally something that works for both the theme install flow and the plugin install flow as @rcrdortiz proposed.

It can be either an inline message, or a modal, or something completely different, whatever you think is best for these new circumstances.

Thank you!
